service/matching/task_reader.go

```diff
@@ -150,8 +150,9 @@ func (tr *taskReader) Stop() {
 				tag.Error(err))
 		}
 		tr.taskGC.RunNow(tr.taskAckManager.GetAckLevel())
-		tr.stopWg.Wait()
 	}
+	// There could be multiple calls to Stop() but all of them should block before task reader is completely stopped.
+	tr.stopWg.Wait()
```

Comment:
I _completely_ agree with this in principle, but this is a rather significant semantic change - as a heads-up for reviewers, it deserves a careful read of all starting and stopping, to make sure it won't misbehave somewhere that stop could be called twice.

Reply:
Oh well, but looking into it further, this might be a problem, but it does not explain the leak. This one is called only once. I'll close this and dig more.